### PR TITLE
test: Cypress | Replace static with Dynamic checks

### PR DESCRIPTION
**Pull request by Aishwarya-U-R**
## Description
- This PR removed static wait from spec - API_with_List_Widget_spec
- Git/GitSync/GitSyncedApps_spec.js (fixes for EE flakyness - WIP - still failing in chrome, pass in electron)
- Remove wait from OpenPaneAndChangeThemeColors() & added n/w check
- Upgrading Cypress from 13.3 to 13.6
- Flaky fix:
      - ClientSide/Widgets/Text/Text2_Spec.ts
      - ClientSide/Widgets/Select/Select2_Spec.ts
      - cypress/e2e/Regression/ClientSide/JSLibrary/Library_spec.ts
      - /ClientSide/Git/GitWithJSLibrary/GitwithCustomJSLibrary_spec.js

#### Type of change
- Script fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [X] Cypress CI runs

#### QA activity:
- [X] Added `Test Plan Approved` label after Cypress tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Enhanced Git synchronization capabilities with improved user interface interactions.
  - Streamlined navigation back to the editor from deployment mode.
  - Implemented more robust table loading checks to ensure data is properly displayed.

- **Bug Fixes**
  - Fixed issues with element presence assertions to improve test reliability.
  - Addressed the visibility of certain UI elements to ensure they appear as expected.

- **Tests**
  - Updated end-to-end test scripts to reflect new interactions and assertions.
  - Added a new test file `GitSyncedApps_spec.js` to cover Git-synced applications.
  - Removed outdated test file `Fork_Template_spec.js`.

- **Refactor**
  - Improved method calls for querying and interacting with elements to enhance test stability.
  - Optimized wait conditions for element visibility to reduce flakiness in tests.

- **Chores**
  - Adjusted the GitHub Actions workflow for client-server build to refine output for matrix counts.
  - Updated CI test workflow to incorporate Electron setup and remove Chrome setup steps.

- **Documentation**
  - Included comments to explain the rationale behind certain timeout values and method changes.

### CODE REVIEW

**Comment by Aishwarya-U-R** [comment]
EE passed runs for [Run1](https://github.com/appsmithorg/appsmith-ee/actions/runs/7225274476), [Run2](https://github.com/appsmithorg/appsmith-ee/actions/runs/7228832394)

File: app/client/cypress/support/Pages/GitSync.ts [hunk 1]
```diff
@@ -356,9 +356,10 @@ export class GitSync {
     assertCreateBranch = true,
   ) {
     if (toUseNewGuid) this.agHelper.GenerateUUID();
-    this.agHelper.AssertElementExist(this._bottomBarCommit);
+    this.agHelper.AssertElementVisibility(this._bottomBarPull);
+    this.agHelper.AssertElementVisibility(this._bottomBarCommit);
     this.agHelper.GetNClick(this._branchButton);
-    this.agHelper.Sleep(2000); //branch pop up to open
+    this.agHelper.AssertElementVisibility(this._branchSearchInput); //branch pop up to open
```
[the remark this answers is not in the document]
Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/GitSync.ts:362]: That Sleep statement is removed. Are you not able to understand the edit made here?

File: app/client/cypress/e2e/Regression/ClientSide/Binding/API_with_List_Widget_spec.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ describe(
           timeout: 20000,
           interval: 1000,
         },
-      ).then(() => cy.wait(500));
+      );
```
[the remark this answers is not in the document]
Comment (mohanarpit) [reply app/client/cypress/e2e/Regression/ClientSide/Binding/API_with_List_Widget_spec.js:66]: @Aishwarya-U-R Nit: Can we just fix this typo along with this PR?

**Review by sarojsarab — APPROVED** [review]

**Pull request merged**